chrisj117/CS-472-GROUP-2-PROJECT
# Setup Docker and Docker Compose

**_WHAT DOES THIS PR DO?_**
- This PR configure docker and docker compose for ease of app development, testing and deployment. It contain both frontend and backend changes.
- It also adds contribution guidelines for open source contribution

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- #11 
- #12 
- #13 

**_HOW DO I TEST OUT THIS PR?_**
- Make sure you have `Docker` and `Docker Compose` installed. The eases way to do this is to install [Docker for Desktop](https://www.docker.com/products/docker-desktop/).
- Clone this repo `git clone` and `cd` to the the project root directory
- Run the command `docker compose -d up` or `docker-compose -d up` and wait for the configuration processes to complete.
- Access the frontend at `http://localhost:5173` and backend at `http://localhost: 8000`. 
NB: The backend is API-based and can be tested using [Postman](https://www.postman.com/) or any REST client of your choice. Check our API [documentation]() for details of endpoints.
